iceoryx_utils/include/iceoryx_utils/internal/cxx/vector.inl

```diff
@@ -51,8 +52,8 @@ inline vector<T, Capacity>::vector(const uint64_t count)
                   << std::endl;
     }
 
-    m_size = std::min(count, Capacity);
-    for (uint64_t i = 0U; i < m_size; ++i)
```

Comment:
Could you please add `cxx::Expects(count <= Capacity);`

Reply:
Sure. Just want to mention that this changes the behavior in case of `count > Capacity` from a warning message to terminate(). For consistency reasons I would also apply the same change to `vector(const uint64_t count, const T& value)`.